test/e2e/scenario/recorder/shadowDom.scenario.ts

```diff
@@ -0,0 +1,242 @@
+import type { RumInitConfiguration } from '@datadog/browser-rum-core'
+import { IncrementalSource, NodeType } from '@datadog/browser-rum/src/types'
+import type { MouseInteractionData, InputData, SerializedNodeWithId } from '@datadog/browser-rum/src/types'
+
+import {
+  createMutationPayloadValidatorFromSegment,
+  findElementWithIdAttribute,
+  findElementWithTagName,
+  findFullSnapshot,
+  findIncrementalSnapshot,
+  findTextContent,
+} from '@datadog/browser-rum/test/utils'
+import type { EventRegistry } from '../../lib/framework'
+import { flushEvents, createTest, bundleSetup, html } from '../../lib/framework'
+import { browserExecute, getBrowserName } from '../../lib/helpers/browser'
+
+/** Will generate the following HTML 
+ * ```html
+ * <my-input-field id="titi">
+ *  #shadow-root
+ *    <div>
+ *      <label  id="label-titi">field titi: </label>
+ *      <input id="input-titi" value="toto">
+ *    </div>
+ *</my-input-field>
+ *```
+ when called like `<my-input-field id="titi" />`
+ */
+const inputShadowDom = `<script>
+ class MyInputField extends HTMLElement {
+   constructor() {
+     super();
+     this.attachShadow({ mode: "open" });
+   }
+   connectedCallback() {
+     const compomentId = this.getAttribute('id') ?? '';
+     const privacyOverride = this.getAttribute("privacy");
+     const parent = document.createElement("div");
+     if (privacyOverride) {
+       parent.setAttribute("data-dd-privacy", privacyOverride);
+     }
+     const label = document.createElement("label");
+     label.setAttribute("id", "label-" + compomentId);
+     label.innerText = "field " + compomentId + ": ";
+     const input = document.createElement("input");
+     input.setAttribute("id", "input-" + compomentId);
+     input.value = "toto"
+     parent.appendChild(label)
+     parent.appendChild(input)
+     this.shadowRoot.appendChild(parent);
+   }
+ }
+       window.customElements.define("my-input-field", MyInputField);
+ </script>
+ `
+
+/** Will generate the following HTML 
+ * ```html
+ * <my-input-field id="titi">
+ *  #shadow-root
+ *    <div>toto</div>
+ *</my-input-field>
+ *```
+ when called like `<my-div />`
+ */
+const divShadowDom = `<script>
+ class CustomDiv extends HTMLElement {
+  constructor() {
+    super();
+    this.attachShadow({ mode: "open" });
+  }
+  connectedCallback() {
+    const div = document.createElement("div");
+    div.textContent = 'toto'
+    this.shadowRoot.appendChild(div);
+  }
+}
+      window.customElements.define("my-div", CustomDiv);
+ </script>
+ `
+
+describe('recorder with shadow DOM', () => {
+  describe('full snapshot', () => {
+    createTest('can overwrite with mask-user-input')
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities.  -->
**💭 thought:** ‏Looking at your test cases description, I'm not sure to get what is the goal of each and if it matches what I would expect to be tested with e2e, meaning:

- mutations inside shadow DOM are collected
- clicks inside shadow DOM are collected with correct target
- change inside shadow DOM are collected with correct target

Could we clarify that?

Reply:
I am not sure what you expect it, can we jump in a call quickly tomorrow?